### PR TITLE
Leverage CTAD with [MD]RangePolicy

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

File: src/details/ArborX_DetailsMinimumSpanningTree.hpp [hunk 1]
```diff
@@ -143,15 +143,13 @@ struct FindComponentNearestNeighbors
     {
       Kokkos::parallel_for(
           "ArborX::MST::find_component_nearest_neighbors_with_lower_bounds",
-          Kokkos::RangePolicy<ExecutionSpace, WithLowerBounds>(space, 0, n),
-          *this);
+          Kokkos::RangePolicy<WithLowerBounds>(space, 0, n), *this);
```
Comment (dalg24) [inline src/details/ArborX_DetailsMinimumSpanningTree.hpp:146]: I think you misunderstood how CTAD works

File: src/details/ArborX_DetailsTreeTraversal.hpp [hunk 2]
```diff
@@ -54,16 +54,13 @@ struct TreeTraversal<BVH, Predicates, Callback, SpatialPredicateTag>
     {
       Kokkos::parallel_for(
           "ArborX::TreeTraversal::spatial::degenerated_one_leaf_tree",
-          Kokkos::RangePolicy<ExecutionSpace, OneLeafTree>(space, 0,
-                                                           predicates.size()),
-          *this);
+          Kokkos::RangePolicy<OneLeafTree>(space, 0, predicates.size()), *this);
```
Comment (dalg24) [inline src/details/ArborX_DetailsTreeTraversal.hpp:57]: Bug

File: src/details/ArborX_DetailsTreeTraversal.hpp [hunk 3]
```diff
@@ -54,16 +54,13 @@ struct TreeTraversal<BVH, Predicates, Callback, SpatialPredicateTag>
...
     }
     else
     {
-      Kokkos::parallel_for("ArborX::TreeTraversal::spatial",
-                           Kokkos::RangePolicy<ExecutionSpace, FullTree>(
-                               space, 0, predicates.size()),
-                           *this);
+      Kokkos::parallel_for(
+          "ArborX::TreeTraversal::spatial",
+          Kokkos::RangePolicy<FullTree>(space, 0, predicates.size()), *this);
```
Comment (dalg24) [inline src/details/ArborX_DetailsTreeTraversal.hpp:63]: Here as well

**Comment by aprokop** [comment]
I may indeed misunderstand some things. I do wonder why it compiled everywhere CPU, though, and not universally failed.

**Comment by dalg24** [comment]
> I may indeed misunderstand some things. I do wonder why it compiled everywhere CPU, though, and not universally failed.

Because only one execution space was enabled and these types were equivalent/compatible.

File: test/tstDetailsDistributor.cpp [hunk 4]
```diff
@@ -61,15 +61,15 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(do_posts_and_waits, DeviceType,
   // receive comm_size packets
   Kokkos::View<int *, DeviceType> exports("Testing::exports", comm_size);
   Kokkos::parallel_for(
-      Kokkos::RangePolicy<ExecutionSpace>(0, comm_size),
+      Kokkos::RangePolicy(0, comm_size),
```
Comment (dalg24) [inline test/tstDetailsDistributor.cpp:64]: Bug
```suggestion
      Kokkos::RangePolicy<ExecutionSpace>(0, comm_size),
```

File: test/tstDetailsDistributor.cpp [hunk 5]
```diff
@@ -61,15 +61,15 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(do_posts_and_waits, DeviceType,
...
       KOKKOS_LAMBDA(int i) { exports(i) = (i < comm_rank ? 0 : comm_rank); });
 
   Kokkos::View<int *, DeviceType> ranks("Testing::ranks", comm_size);
   ArborX::Details::KokkosExt::iota(ExecutionSpace{}, ranks, 0);
 
   Kokkos::View<int *, DeviceType> imports_ref("Testing::v_ref", comm_size);
   Kokkos::parallel_for(
-      Kokkos::RangePolicy<ExecutionSpace>(0, comm_size), KOKKOS_LAMBDA(int i) {
+      Kokkos::RangePolicy(0, comm_size), KOKKOS_LAMBDA(int i) {
```
Comment (dalg24) [inline test/tstDetailsDistributor.cpp:72]: Here too
```suggestion
      Kokkos::RangePolicy<ExecutionSpace>(0, comm_size), KOKKOS_LAMBDA(int i) {
```

File: test/tstDetailsTreeConstruction.cpp [hunk 6]
```diff
@@ -130,8 +130,7 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(indirect_sort, DeviceType, ARBORX_DEVICE_TYPES)
   Kokkos::View<unsigned int *, DeviceType> k("k", n);
   // Fill K with 4, 3, 2, 1
   FillK<DeviceType> fill_k_functor(k);
-  Kokkos::parallel_for("fill_k", Kokkos::RangePolicy<ExecutionSpace>(0, n),
-                       fill_k_functor);
+  Kokkos::parallel_for("fill_k", Kokkos::RangePolicy(0, n), fill_k_functor);
```
Comment (dalg24) [inline test/tstDetailsTreeConstruction.cpp:133]: Bug

**Comment by dalg24** [comment]
I stopped and didn't actually go through the rest of the changes.  I expect you will find other occurrences if any

**Comment by aprokop** [comment]
> I stopped and didn't actually go through the rest of the changes. I expect you will find other occurrences if any

Thank you. I used regex and should have been much more careful about the changes. I'm sorry.

**Comment by aprokop** [comment]
Rebased.

**Pull request merged**